### PR TITLE
Remove TODO from public documentation

### LINKED ISSUE #47366: Internal code comments are put in triple-slash doc comments
https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.syntaxnode.findnode?view=roslyn-dotnet

> Remarks
> TODO: This should probably be reimplemented with ChildThatContainsPosition(Int32)

I don't believe such a comment should be a doc comment.

There will probably be more occurrences of this. At least, I'd search with `///.+?TODO` regex in the repo to detect some of these issues.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #47366

There are more TODOs in triple-slash doc comments. But I only found this one in a public API.

### CODE REVIEW

**Comment by Youssef1313** [comment]
> 
> 
> Rather than remove the semantic reference (the `<see>` element), you can change `<remarks>` to `<devdoc>` which continues to be validated by the compiler but doesn't show in rendered documentation.

Thanks. I didn't know about the existence of devdoc.

File: src/Compilers/Core/Portable/Syntax/SyntaxNode.cs [hunk 1]
```diff
@@ -833,9 +833,9 @@ public IEnumerable<SyntaxNodeOrToken> DescendantNodesAndTokensAndSelf(TextSpan s
         /// If <paramref name="getInnermostNodeForTie"/> is true, then it returns lowest descending node encompassing the given <paramref name="span"/>.
         /// Otherwise, it returns the outermost node encompassing the given <paramref name="span"/>.
         /// </summary>
-        /// <remarks>
+        /// <devdoc>
         /// TODO: This should probably be reimplemented with <see cref="ChildThatContainsPosition"/>
-        /// </remarks>
+        /// </devdoc>
```
Comment (sharwell) [inline src/Compilers/Core/Portable/Syntax/SyntaxNode.cs:838]:
```suggestion
        /// </devdoc>
```

**Comment by sharwell** [comment]
>  I didn't know about the existence of devdoc.

It's a made-up XML element historically used for this purpose. The documentation compiler defines several well-known elements, but tools are allowed to define and/or use others.

**Comment by Youssef1313** [comment]
@sharwell I found that a devdoc was used in dotnet/runtime but was **included** in the public documentation:

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ThreadWaitReason.cs#L6-L9

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.threadwaitreason?view=netcore-3.1

The XML for the doc page include the exact same text as summary.

Any idea why this is happening?

**Comment by sharwell** [comment]
Here's the XML documentation for `ThreadWaitReason`:
https://github.com/dotnet/dotnet-api-docs/blob/78344cc2feabc5795757fe8770e7692c4f9d4211/xml/System.Diagnostics/ThreadWaitReason.xml#L80

The `<devdoc>` in dotnet/runtime is just for local reference but isn't the actual API documentation for this type.

**Comment by jcouv** [comment]
Squashed/merged. Thanks @Youssef1313 !

**Pull request merged**